app/models/mappers/check-activities.js

```diff
@@ -11,18 +169,211 @@ const getActivityLabelFromEvent = (event) => {
+ * @returns {string}
+ */
+
+/**
+ * @param {string} eventType
+ * @returns {GetActivityLabelFromAuditFieldRecordFn}
+ */
+const getActivityLabelFromAuditFieldRecord = (eventType) => (auditFieldRecord) => {
+  const [fieldValue] = auditFieldRecord
+
+  if (activityLabels[fieldValue]) {
+    return `${activityLabels[fieldValue]} ${eventType}`
+  }
+  return 'N/A'
+}
+
+/**
+ *
+ * @param {CreatedDogEvent} createdDogEvent
+ * @returns {string}
+ */
+const getActivityLabelFromCreatedDog = (createdDogEvent) => {
+  return `Dog record created (${createdDogEvent.status.status})`
+}
+
+/**
+ * @param {ChangeEvent} event
+ * @returns {ActivityRow[]}
+ */
+const mapAuditedChangeEventToCheckActivityRows = (event) => {
+  const activityRowInfo = getDateAndTeamMemberFromEvent(event)
+  const auditedFieldRecords = event.changes.edited
+  /**
+   * @type {ActivityRow[]}
+   */
+  const activityRows = []
+
+  return auditedFieldRecords.reduce((activityRows, changeRecord) => {
+    let changeType = addedEvents.includes(changeRecord[0]) ? 'added' : 'updated'
+
+    if (changeRecord[0] === 'status') {
+      const [,, statusName] = changeRecord
+      changeType = statusName
+    }
+    const activityLabel = getActivityLabelFromAuditFieldRecord(changeType)(changeRecord)
+
+    if (filterSameDate(changeRecord) && activityLabel !== 'N/A') {
+      const activityRow = { ...activityRowInfo, activityLabel }
+      return [...activityRows, activityRow]
+    }
+
+    return [...activityRows]
+  }, activityRows)
+}
+
+/**
+ * @param {DDIEvent} event
+ * @returns {ActivityRow}
+ */
+const mapActivityDtoToCheckActivityRow = (event) => {
+  return {
+    activityLabel: getActivityLabelFromEvent(event),
+    ...getDateAndTeamMemberFromEvent(event)
+  }
+}
+
+/**
+ * @param {CreatedEvent} event
+ * @returns {ActivityRow[]}
+ */
+const mapCreatedEventToCheckActivityRows = (event) => {
+  const dateAndTeamMemberData = getDateAndTeamMemberFromEvent(event)
+
+  return event.created.dogs.map(createdDogEvent => {
+    return {
+      activityLabel: getActivityLabelFromCreatedDog(createdDogEvent),
+      ...dateAndTeamMemberData
+    }
+  })
+}
+
+/**
+ * @param {DDIEvent[]} events
+ * @returns {ActivityRow[]}
+ */
+const flatMapActivityDtoToCheckActivityRow = (events) => {
+  /**
+   * @type {ActivityRow[]}
+   */
+  const activityRowsAccumulator = []
+  return events.reduce((activityRows, event) => {
+    /**
+     * @type {ActivityRow[]}
+     */
+    const addedRows = []
+
+    if (event.type === 'uk.gov.defra.ddi.event.activity') {
+      addedRows.push(mapActivityDtoToCheckActivityRow(event))
+    }
+
+    if (event.type === 'uk.gov.defra.ddi.event.update') {
+      addedRows.push(...mapAuditedChangeEventToCheckActivityRows(event))
+    }
+
+    if (event.type === 'uk.gov.defra.ddi.event.create') {
+      addedRows.push(...mapCreatedEventToCheckActivityRows(event))
+    }
+
+    return [...activityRows, ...addedRows]
+  }, activityRowsAccumulator)
+}
+
```

Comment:
This is good code but might be easier if it were split as quite a long file. Perhaps one for later